.github/workflows/e2e-tests.yml

```diff
@@ -16,9 +16,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v1
-      - uses: actions/setup-go@v2
+      - uses: actions/setup-go@v3
         with:
-          go-version: '~1.18'
+          go-version-file: "go.mod"
```

Comment:
Note: this should help reduce maintenance burden over time. Most of the next gen repositories (e.g. rukpak, deppy, etc.) have been using this approach for a bit.